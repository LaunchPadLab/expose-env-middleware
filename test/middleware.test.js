const request = require('supertest')
const middleware = require('../middleware')
const express = require('express')

const ENV = { foo: 'bar' }
const GET_ENV = () => ENV

describe('middleware', () => {
  it('requires a getEnv() function', () => {
    const app = express()
    expect(() => app.use(middleware())).toThrow()
  })
  it('exposes a stringified environment', () => {
    const app = express()
    app.use(middleware(GET_ENV))
    return request(app)
      .get('/env.js')
      .then(res => expect(res.text).toMatchSnapshot())
  })
  it('accepts custom filename argument', () => {
    const app = express()
    app.use(middleware(GET_ENV, { filename: 'foo' }))
    return request(app)
      .get('/foo')
      .expect(200)
  })
  it('accepts custom template argument', () => {
    const app = express()
    app.use(middleware(GET_ENV, { template: env => 'ur env is ' + JSON.stringify(env) }))
    return request(app)
      .get('/env.js')
      .then(res => expect(res.text).toMatchSnapshot())
  })
})
