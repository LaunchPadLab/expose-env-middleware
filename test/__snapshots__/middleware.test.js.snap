// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`middleware accepts custom template argument 1`] = `"ur env is {\\"foo\\":\\"bar\\"}"`;

exports[`middleware exposes a stringified environment 1`] = `
"
    window.process = {
      env: {\\"foo\\":\\"bar\\"}
    }  
  "
`;
